src/sidebar/services/stream-filter.js

```diff
@@ -1,3 +1,46 @@
+/**
+ * @typedef {'equals'|'one_of'} Operator
+ */
+
+/**
+ * Filter clause against which annotation updates are tested before being
+ * sent to the client.
+ *
+ * @typedef FilterClause
+ * @prop {string} field
+ * @prop {Operator} operator
+ * @prop {any} value
+ * @prop {boolean} case_sensitive - TODO: Backend doesn't use this at present,
+ *   but it seems important for certain fields (eg. ID).
+ */
+
+/**
+ * @typedef Filter
+ * @prop {string} match_policy - TODO: Remove this, the backend doesn't use it any more.
+ * @prop {FilterClause[]} clauses
+ * @prop {Object} actions - TODO: Remove this, the backend doesn't use it any more.
+ * @prop {boolean} [actions.create]
```

Comment:
I realize this is `todo remove`, but in general If these are props of actions, I would prefer (if possible) 2 spaces to help visually see they belong to actions rather than than FIlter. that would help clarify nested structures for me.

```
 * @prop {Object} actions - TODO: Remove this, the backend doesn't use it any more.
 *   @prop {boolean} [actions.create]
 *   @prop {boolean} [actions.update]
 *   @prop {boolean} [actions.delete]
```

Reply:
I like this. Will do.